mk/compile.mk

```diff
@@ -146,3 +148,76 @@ $(foreach f, $(gen-srcs), $(eval $(call \
 	process_srcs,$(out-dir)/$(f),$(out-dir)/$(base-prefix)$$(basename $f).o)))
 
 $(objs): $(conf-file)
+
+define gen-defines-file
+# c-filename in $1
+# h-filename in $2
+# s-filename in $3
```

Comment:
The `.s` file is an intermediate file, as such it doesn't need to be in the parameter list. It would be better to generate its path internally from $2.